File: cplusplus/06-list-of-strings/list_of_strings.cpp
#include <iostream>
#include <list>
#include <string>

using namespace std;

void printlist(list <string> userlist){
    cout << "User entered strings: ";
    std::list<string>::iterator it;
    string tmpstr; 
    for (it = userlist.begin(); it != userlist.end(); it++){
        cout << *it << " "; 
    }
    cout << endl; 
}


int main(){
    string input;
    list <string> userlist; 

    while (input != "-1"){
	cout << "Enter a string (Enter -1 to quit): ";
	cin >> input;
	userlist.push_back(input);
	printlist(userlist);
    }
    cout << "Quitting";
}

File: java/06-list_of_strings/list_of_strings.java
import java.util.*;

public class listofstrings
{
    public static void main(String[] args)
    {
	String inputstr = "";
	Scanner scanner = new Scanner(System.in);
	List<String> userlist = new ArrayList<String>();
	
	while (!inputstr.equals("-1")){
	    System.out.println("Enter a string (Enter -1 to quit): ");
	    inputstr = scanner.nextLine().strip();
	    userlist.add(inputstr);
	    System.out.println("User entered strings: " + userlist);
	}
    }
}

